src/strategies/logs-to-subgraph/transformer.mjs

```diff
@@ -0,0 +1,69 @@
+// @format
+import fs from "fs";
+
+import logger from "../../logger.mjs";
+import { parseJSON } from "../../utils.mjs";
+
+export const name = "logs-to-subgraph";
+const log = logger(name);
+export const version = "0.1.0";
+
+export function onClose() {
+  log("closed");
+  return {
+    write: null,
+    messages: [],
+  };
+}
+
+export function onError(error) {
+  log(error.toString());
+  throw error;
+}
+
+let validatedContracts;
+
+export function onLine(line, contractsPath) {
+  const contracts = JSON.parse(fs.readFileSync(contractsPath));
```

Comment:
considering that this function is called very often, doing a readFileSync is every time is inefficient

Reply:
I agree. It is not efficient but since `onLine` isn't async I couldn't use `fs.readFile`.